front/components/providers/AI21Setup.jsx

```diff
@@ -52,7 +52,7 @@ export default function AI21Setup({ owner, open, setOpen, config, enabled }) {
     });
     let data = await res.json();
     setEnableRunning(false);
-    mutate(`/api/w/${owner.sId}/providers`);
+    await mutate(`/api/w/${owner.sId}/providers`);
```

Comment:
This await is going to delay `setOpen(false)`, are we OK with that ?

Reply:
Should we move the setOpen() before the awaits then?

Reply:
I moved all the `setOpen()` before the `await mutate`, which I think is going to address your concern.